roles/dnsmasq/tasks/main.yml

```diff
@@ -77,6 +77,6 @@
   wait_for:
     host: "{{dns_server}}"
     port: 53
-    delay: 5
-  when: inventory_hostname == groups['kube-node'][0]
+    timeout: 180
+  when: inventory_hostname == groups['kube-node'][0] and groups['kube-node'][0] in play_hosts
```

Comment:
play_hosts has been deprecated in 2.2, it was the same as the new ansible_play_batch variable.

http://docs.ansible.com/ansible/playbooks_variables.html